source/Mole-Tests/GraphTest.class.st

```diff
@@ -14,65 +14,54 @@ GraphTest class >> isAbstract [
 ]
 
 { #category : #private }
-GraphTest >> edgeOfInvalidType [
+GraphTest >> assertEdgesIncidentTo: aVertex in: aGraph match: aDictionary [
 
-	^ self subclassResponsibility
+	| labelByEdge edges |
+
+	labelByEdge := Dictionary new.
+	edges := aGraph edgesIncidentTo: aVertex.
+	edges
+		do: [ :edge | 
+			edge
+				withLabelDo: [ :label | labelByEdge at: edge asString put: label ]
+				ifUnlabeled: [ self fail ] ].
+	self assertCollection: labelByEdge keys hasSameElements: aDictionary keys.
+	aDictionary
+		keysAndValuesDo: [ :edgeString :weight | self assert: (labelByEdge at: edgeString) equals: weight ]
```

Comment:
This kind of test can be non-deterministic because the string you create in the match dictionary are fixed, and the other ones are generated here. The endpoints in the undirected edge ends up in a Bag or Set, so the order can be different, then the string also can be different.